### PR TITLE
arm64: dts: qcom: Add msm8916-acer-a1-724 and msm8916-gplus-fl8005a

### PR DESCRIPTION
**Pull request by wonderfulShrineMaidenOfParadise**
Prepare for upstreaming.
- [x] https://lore.kernel.org/lkml/20221123033524.148682-1-linmengbo0689@protonmail.com/
- [x] https://lore.kernel.org/lkml/20230107132932.139669-1-linmengbo0689@protonmail.com/

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+};
+
+&sdhc_2 {
+	status = "okay";
+
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&sdc2_clk_on &sdc2_cmd_on &sdc2_data_on>;
+	pinctrl-1 = <&sdc2_clk_off &sdc2_cmd_off &sdc2_data_off>;
+
+	cd-gpios = <&msmgpio 38 GPIO_ACTIVE_HIGH>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts]: Should be low in downstream but it won't be enabled in mainline.

File: arch/arm64/boot/dts/qcom/msm8916-gplus-fl8005a.dts [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+#include "msm8916-pm8916.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	model = "GPLUS FL8005A";
+	compatible = "gplus,fl8005a", "qcom,msm8916";
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-gplus-fl8005a.dts]: The manufacturer should be Foxda. But there is no other known device with similar design.

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+ * NOTE: The original firmware from Acer can only boot 32-bit kernels.
+ * To boot this device tree using arm64 it is necessary to flash 64-bit
+ * TZ/HYP firmware (e.g. taken from the DragonBoard 410c).
+ * See https://wiki.postmarketos.org/wiki/Acer_Iconia_Talk_S_(acer-a1-724)
+ * for suggested installation instructions.
+ */
+
+/ {
+	model = "Acer Iconia Talk S A1-724";
+	compatible = "acer,a1-724", "qcom,msm8916";
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts]: https://remont-aud.net/load/drugaja_apparatura/planshety_ehl_knigi_fotoramki/quanta_alt_alte_v4_gold_rev_3a_acer_iconia_talk_agusta_s_a1_724/288-1-0-52147
something like `quanta,agusta`? Need more info.

**Review by stephan-gh — DISMISSED** [review]
Looks good, feel free to send it upstream :+1:

**Comment by wonderfulShrineMaidenOfParadise** [comment]
Upstream
https://lore.kernel.org/lkml/20221119194340.91789-1-linmengbo0689@protonmail.com/
https://lore.kernel.org/lkml/20221119202316.93142-1-linmengbo0689@protonmail.com/

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts (hunk 3 above)
Comment (wonderfulShrineMaidenOfParadise) [reply arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts]: https://fcc.report/FCC-ID/HFS-FT7
This tablet is almost the same device, except it has flashlight and ambient light/proximity sensor.

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+	reserved-memory {
+		mpss_mem: mpss@86800000 {
+			reg = <0x0 0x86800000 0x0 0x4e00000>;
+			no-map;
+		};
+
+		gps_mem: gps@8b600000 {
+			reg = <0x0 0x8b600000 0x0 0x200000>;
+			no-map;
+		};
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts]: Wrong commit.

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&sdc2_clk_on &sdc2_cmd_on &sdc2_data_on>;
+	pinctrl-1 = <&sdc2_clk_off &sdc2_cmd_off &sdc2_data_off>;
+
+	cd-gpios = <&msmgpio 38 GPIO_ACTIVE_HIGH>;
+
+	status = "okay";
+};
+
+&sound {
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts:191]: Microphones are broken.

File: arch/arm64/boot/dts/qcom/msm8916-gplus-fl8005a.dts [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&sdc2_clk_on &sdc2_cmd_on &sdc2_data_on>;
+	pinctrl-1 = <&sdc2_clk_off &sdc2_cmd_off &sdc2_data_off>;
+
+	cd-gpios = <&msmgpio 38 GPIO_ACTIVE_LOW>;
+
+	status = "okay";
+};
+
+&sound {
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-gplus-fl8005a.dts:237]: Jack microphone is broken.

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts [hunk 7]
```diff
@@ -109,6 +109,34 @@
 	status = "okay";
 };
 
+&dsi0 {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&mdss_default>;
+	pinctrl-1 = <&mdss_sleep>;
+
+	panel@0 {
+		compatible = "acer,hx8394d";
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts]: would expect a power supply here, maybe l17 if there is no gpio etc listed?

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+&usb_hs_phy {
+	extcon = <&usb_id>;
+};
+
+&wcd_codec {
+	qcom,micbias-lvl = <2800>;
+	qcom,mbhc-vthreshold-low = <150 150 500 500 500>;
+	qcom,mbhc-vthreshold-high = <150 150 500 500 500>;
+	qcom,hphl-jack-type-normally-open;
+};
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts:213]: Volume+, Volume - and Pause buttons are good now. The headphone microphone is still broken.
The values are to be verified.

File: arch/arm64/boot/dts/qcom/msm8916-gplus-fl8005a.dts [hunk 9]
```diff
@@ -0,0 +1,448 @@
...
+&usb_hs_phy {
+	extcon = <&usb_id>;
+};
+
+&wcd_codec {
+	qcom,micbias-lvl = <2800>;
+	qcom,mbhc-vthreshold-low = <150 180 237 450 500>;
+	qcom,mbhc-vthreshold-high = <150 180 237 450 500>;
+	qcom,hphl-jack-type-normally-open;
+};
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-gplus-fl8005a.dts:267]: Volume+, Volume - and Pause buttons are good now. The headphone microphone sometimes works, sometimes doesn't.
The values are to be verified.

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts (hunk 7 above)
Comment (wonderfulShrineMaidenOfParadise) [reply arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts]: I disabled l17 and the panel doesn't display anything in userspace. Makes sense.

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts [hunk 10]
```diff
@@ -0,0 +1,382 @@
...
+
+&blsp_i2c2 {
+	status = "okay";
+
+	accelerometer@10 {
+		compatible = "bosch,bmc150_accel";
+		reg = <0x10>;
+		interrupt-parent = <&msmgpio>;
+		interrupts = <115 IRQ_TYPE_EDGE_RISING>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts]: Should be `IRQ_TYPE_EDGE_FALLING` instead? `0x2002` in downstream.

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts [hunk 11]
```diff
@@ -0,0 +1,382 @@
...
+			       "-1", "0", "0",
+			       "0", "0", "1";
+	};
+
+	magnetometer@12 {
+		compatible = "bosch,bmc150_magn";
+		reg = <0x12>;
+
+		interrupt-parent = <&msmgpio>;
+		interrupts = <69 IRQ_TYPE_EDGE_RISING>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts:93]: To be verified.

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts (hunk 10 above)
Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts]: For BMC150, the interrupt type can be configured to high/low/falling/rising in the hardware. This means the mainline driver counts and not what downstream does. bmc150-accel-core.c specifically requests

https://github.com/msm8916-mainline/linux/blob/ca57f02295f188d6c65ec02202402979880fa6d8/drivers/iio/accel/bmc150-accel-core.c#L1762

File: arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts [hunk 12]
```diff
@@ -0,0 +1,382 @@
...
+
+		vdd-supply = <&pm8916_l17>;
+		vddio-supply = <&pm8916_l6>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&accel_int_default>;
+
+		mount-matrix = "0", "-1", "0",
+			       "-1", "0", "0",
+			       "0", "0", "1";
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-acer-a1-724.dts]: Maybe the same value for the magetometer? To be verified

**Comment by wonderfulShrineMaidenOfParadise** [comment]
Aborting fl8005a mainlining due to lacking of hardware.

**Pull request closed without merging**